src/plugins/MoreSearchOperators.ts

```diff
@@ -0,0 +1,213 @@
+    ANSWER_SORT_ORDER: {
+        componentType: "ANSWER",
+        regex: /\s*(asc|desc)(?:ending)?/i,
+        validator: function (match) {
+            let value = match.getMatch(1);
+            switch (value) {
+                case "asc":
+                    match.setData("sortOrder", "asc");
+                    return true;
+                case "desc":
+                    match.setData("sortOrder", "desc");
+                    return true;
+                default:
+                    return false;
+            }
+        },
+        follows: ["FILTER_SORT_ORDER"],
+        queryKey: "sort_order",
+        mutable: true,
+        _dataKey: "sortOrder"
+    },
+    ANSWER_SORT_BY: {
+        componentType: "ANSWER",
+        regex: /\s*(relevance|timestamp)/i,
+        validator: function (match) {
+            let value = match.getMatch(1);
+            switch (value) {
+                case "relevance":
+                    match.setData("sortBy", "relevance");
+                    return true;
+                case "timestamp":
+                    match.setData("sortBy", "timestamp");
+                    return true;
+                default:
+                    return false;
+            }
+        },
+        follows: ["FILTER_SORT_BY"],
+        queryKey: "sort_by",
+        mutable: true,
+        _dataKey: "sortBy"
+    },
+    ANSWER_FILE_NAME: {
+        componentType: "ANSWER",
+        regex: /(?:\s*([^\s]+))/,
+        validator: function (match) {
+            match.setData("fileName", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_FILE_NAME"],
+        queryKey: "attachment_filename",
+        mutable: true,
+        _dataKey: "fileName"
+    },
+    ANSWER_FILE_TYPE: {
+        componentType: "ANSWER",
+        regex: /(?:\s*([^\s]+))/,
+        validator: function (match) {
+            match.setData("fileType", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_FILE_TYPE"],
+        queryKey: "attachment_extension",
+        mutable: true,
+        _dataKey: "fileType"
+    },
+    ANSWER_EMBED_TYPE: {
+        componentType: "ANSWER",
+        regex: /\s*([^\s]+)/i,
+        validator: function (match) {
+            match.setData("embedType", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_EMBED_TYPE"],
+        queryKey: "embed_type",
+        mutable: true,
+        _dataKey: "embedType"
+    },
+};
+
+export default definePlugin({
+    name: "MoreSearchOperators",
+    description: "Adds experimental search operators.",
+    authors: [Devs.Davri],
+
+    patches: [{
+        find: "Messages.SEARCH_SHORTCUT",
+        replacement: {
+            match: /(.)\((.),.\..{1,3}\.ANSWER_PINNED/,
+            replace: Object.keys(searchOperators).map(a => `$1($2,"${a}",$self.searchOperators.${a}),`).join("") + Object.keys(searchAnswers).map(a => `$1($2,"${a}",$self.searchAnswers.${a}),`).join("") + "$&"
```

Comment:
```suggestion
            match: /(\i)\((\i),\i\.\i\.ANSWER_PINNED/,
            replace: Object.keys(searchOperators).map(a => `$1($2,"${a}",$self.searchOperators.${a}),`).join("") + Object.keys(searchAnswers).map(a => `$1($2,"${a}",$self.searchAnswers.${a}),`).join("") + "$&"
```
Vencord has a custom regex escape `\i` that matches SWC's mangled variable names

Reply:
Didn't know about that, it should probably be added to the plugin guide